nodebuilder/das/config.go

```diff
@@ -0,0 +1,70 @@
+package das
+
+import (
+	"errors"
+	"time"
+)
+
+// Config contains configuration parameters for the DASer (or DASing process)
+type Config struct {
+	//  SamplingRange is the maximum amount of headers processed in one job.
+	SamplingRange uint
+
+	// ConcurrencyLimit defines the maximum amount of sampling workers running in parallel.
+	ConcurrencyLimit int
+
+	// BackgroundStoreInterval is the period of time for background checkpointStore to perform a checkpoint backup.
+	BackgroundStoreInterval time.Duration
+
+	// PriorityQueueSize defines the size limit of the priority queue
+	PriorityQueueSize int
+
+	// GenesisHeight is the height sampling will start from
+	GenesisHeight uint
+}
+
+// TODO(@derrandz): parameters needs performance testing on real network to define optimal values
+func DefaultConfig() Config {
+	return Config{
+		SamplingRange:           100,
+		ConcurrencyLimit:        16,
+		BackgroundStoreInterval: 10 * time.Minute,
+		PriorityQueueSize:       16 * 4,
+		GenesisHeight:           1,
+	}
+}
+
+// Validate performs basic validation of the config.
+//
+// SamplingRange == 0 will cause the jobs' queue to be empty
+//
+//	therefore no sampling jobs will be reserved and more importantly
+//	 the DASer will break
+//
+// ConcurrencyLimit == 0 will cause the number of workers to be 0 and thus
+//
+//	no threads will be assigned to the waiting jobs therefore breaking the DASer
+//
+// OR GenesisHeight == 0 there is no block height 0 to be sampled thus will break the DASer.
+//
+// # On the other hand
+//
+// BackgroundStoreInterval == 0 disables background storer,
+// PriorityQueueSize == 0 disables prioritization of recently produced blocks for sampling
+//
+// Both of which won't break the DASer
+func (cfg *Config) Validate() error {
```

Comment:
We also don't need validation since we won't provide a Config here, it would just point to das.Params

Reply:
🤔 hmmm, I see what you are saying, and I agree, but for the record, if we wanted to be very strict with our patterns, here are two arguments of why you would keep this redundant layer of validation:

1. Validation in this level concerns `nodebuilder`, meaning that this is validation for values coming from a file or CLI, irrespective of where it will go (_i.e: the underlying implementation of `das` that will receive this configuration_), so we must keep it to respect the pattern. To drive this point home, we could use a different daser down the line (_we won't but you see my point_)

2. The validation of `das.paramaeters` in the functional options inside of the `das` package was made in an effort to further isolate the package from `nodebuilder` and make it more of a standalone (_per my understand, that's the purpose we are doing this, correct me if I am wrong_) thus, that validation is useful for any code that will invoke those functional parameters. For the case of nodebuilder, it just so happens that it already validates its own configuration, but for other users (_when we separate das into its own go package_) it might not be the case.

Reply:
After some thought, we need to keep validation at this level, but change it slightly.

The reason is, we want `das.Config` to reflect in `das.Parameters` of the daser when `das.Config` has non-zero values retrieved from CLI or file or API, this is due to [this piece of code](https://github.com/celestiaorg/celestia-node/pull/1225/files#diff-9eef3a4c09b5618e9b60c229170908fc389f8a28ce84e29e47c82d33b74e32b2R23-R28) that retrieves configuration values and "passes them as options".

However this will cause us to change the [functional options validation ](https://github.com/celestiaorg/celestia-node/pull/1225/files#diff-fa4e1a1d6adb3bac7ebcb4524081bfce3ae7764b2c0a249c8a5655cdcac5b3bdR55-R64) to not return errors and cause the constructor to fail but rather to use default options when validation fails, and log a warning line.

I will hold until we reach a decision on this.

@renaynay @Wondertan

Reply:
I've settled on:

1. Deduping the code by using `type Config das.Parameters`
2. Keeping the `(*Config) Validate` to avoid the impact of removing it that is described in [the second comment](https://github.com/celestiaorg/celestia-node/pull/1225#discussion_r1004365160) based on argumentation from [the first comment](https://github.com/celestiaorg/celestia-node/pull/1225#discussion_r1004333336)

Reply:
@derrandz, sorry for the confusion. We didn't want to avoid the second validation, but only dedup the code and keep it grouped where it logically should be in `das`. So the point where you settled is exactly what we discussed.

Reply:
Clear, I will do one more iteration on this.

Reply:
@renaynay @Wondertan I think this is brilliant, thanks for the comments